contracts/Liquidator/Liquidator.sol

```diff
@@ -334,8 +377,17 @@ contract Liquidator is Ownable2StepUpgradeable, ReentrancyGuardUpgradeable, Acce
         if (!vTokenCollateral.transfer(msg.sender, theirs)) {
             revert VTokenTransferFailed(address(this), msg.sender, theirs);
         }
-        if (!vTokenCollateral.transfer(treasury, ours)) {
-            revert VTokenTransferFailed(address(this), treasury, ours);
+
+        if (IVBep20(address(vTokenCollateral)).redeem(ours) != 0) {
+            pendingRedeem.push(address(vTokenCollateral));
+        } else {
+            address underlying = IVBep20(address(vTokenCollateral)).underlying();
+            uint256 underlyingBalance = IVToken(underlying).balanceOf((address(this)));
+            if (!IVToken(underlying).transfer(protocolShareReserve, underlyingBalance)) {
+                revert UnderlyingTransferFailed(address(vTokenCollateral), underlying);
+            }
+            IProtocolShareReserve(protocolShareReserve).updateAssetsState(address(comptroller), underlying, 1);
+            emit ReservesReduced(msg.sender, address(vTokenCollateral), ours);
```

Comment:
I would use functions to reuse this sequence of statements in the `_reduceReservesInternal()`